spec/faraday/response/raise_error_spec.rb

```diff
@@ -3,26 +3,74 @@
     Faraday.new do |b|
       b.response :raise_error
       b.adapter :test do |stub|
-        stub.get('ok')        { [200, {'Content-Type' => 'text/html'}, '<body></body>'] }
-        stub.get('not-found') { [404, {'X-Reason' => 'because'}, 'keep looking'] }
-        stub.get('error')     { [500, {'X-Error' => 'bailout'}, 'fail'] }
+        stub.get('ok') { [200, { 'Content-Type' => 'text/html' }, '<body></body>'] }
+        stub.get('bad-request') { [400, { 'X-Reason' => 'because' }, 'keep looking'] }
+        stub.get('unauthorized') { [401, { 'X-Reason' => 'because' }, 'keep looking'] }
+        stub.get('forbidden') { [403, { 'X-Reason' => 'because' }, 'keep looking'] }
+        stub.get('not-found') { [404, { 'X-Reason' => 'because' }, 'keep looking'] }
+        stub.get('proxy-error') { [407, { 'X-Reason' => 'because' }, 'keep looking'] }
+        stub.get('unprocessable-entity') { [422, { 'X-Reason' => 'because' }, 'keep looking'] }
+        stub.get('4xx') { [499, { 'X-Reason' => 'because' }, 'keep looking'] }
+        stub.get('server-error') { [500, { 'X-Error' => 'bailout' }, 'fail'] }
       end
     end
   end
 
-  it 'raises no exceptio for 200 responses' do
+  it 'raises no exception for 200 responses' do
     expect { conn.get('ok') }.not_to raise_error
   end
 
-  it 'raise Faraday::Error::ResourceNotFound for 404 responses' do
-    expect { conn.get('not-found') }.to raise_error(Faraday::Error::ResourceNotFound) do |ex|
+  it 'raise Faraday::ResourceNotFound for 400 responses' do
```

Comment:
The description could use the “raises” tense of that verb (throughout these cases). It was the wrong form before your changes, too.